File: python/Number of Steps to Reduce a Number in Binary Representation to One.py
def divide_by_two( s):
    s.pop()


def add_one( s):
    i = len(s) - 1

    # Iterating while the character is 1 and changing to 0
    while i >= 0 and s[i] != "0":
        s[i] = "0"
        i -= 1

    if i < 0:
        s.insert(0, "1")
    else:
        s[i] = "1"


def numSteps(s: str) -> int:
    s = list(s)
    operations = 0

    while len(s) > 1:
        N = len(s)

        if s[N - 1] == "0":
            divide_by_two(s)
        else:
            add_one(s)

        operations += 1

    return operations
print(numSteps("1101"))

File: JAVA/java/src/NumberofStepstoReduceaNumberinBinaryRepresentationtoOne.java
public class NumberofStepstoReduceaNumberinBinaryRepresentationtoOne {
    public int numSteps(String s) {
        int N = s.length();

        int operations = 0;
        int carry = 0;
        for (int i = N - 1; i > 0; i--) {
            int digit = Character.getNumericValue(s.charAt(i)) + carry;

            if (digit % 2 == 1) {
                operations += 2;
                carry = 1;
            } else {
                operations++;
            }
        }

        return operations + carry;
    }

    public static void main(String[] args) {
        NumberofStepstoReduceaNumberinBinaryRepresentationtoOne ob=new NumberofStepstoReduceaNumberinBinaryRepresentationtoOne();
        System.out.println(ob.numSteps("1101"));
    }
}
